src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,353 @@
+        {
+            Debug.Assert(possiblyContainingOriginalType.IsDefinition);
+            Debug.Assert(type != null);
+
+            for (var current = type.OriginalDefinition; current != null; current = current.ContainingType)
+            {
+                if (SameOriginalNamedType(current, possiblyContainingOriginalType))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool InheritsFromIgnoringConstruction(this ITypeSymbol type, INamedTypeSymbol baseType)
+        {
+            Debug.Assert(type.IsDefinition);
+            Debug.Assert(baseType.IsDefinition);
+
+            for (var current = type; current != null; current = current.BaseType?.OriginalDefinition)
+            {
+                if (SameOriginalNamedType(baseType, current as INamedTypeSymbol))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool SameOriginalNamedType(INamedTypeSymbol t1, INamedTypeSymbol t2)
+        {
+            Debug.Assert(t1 != null);
+            Debug.Assert(t1.IsDefinition);
+            Debug.Assert(t2?.IsDefinition != false);
+
+            // We expect a given named type definition to satisfy reference identity.
+            if (t1 == t2)
+            {
+                return true;
+            }
+
+            // We relax this to permit a type to be represented by multiple symbols (e.g. separate compilations)
+            return t2 != null && t1.Name == t2.Name && t1.Arity == t2.Arity && SameOriginalSymbol(t1.ContainingSymbol, t2.ContainingSymbol);
+        }
+
+        private static bool SameOriginalSymbol(ISymbol s1, ISymbol s2)
+        {
+            if (s1 == s2)
+            {
+                return true;
+            }
+
+            if (s1 == null || s2 == null)
+            {
+                return false;
+            }
+
+            switch (s1.Kind)
+            {
+                case SymbolKind.NamedType:
+                    return SameOriginalNamedType(s1 as INamedTypeSymbol, s2 as INamedTypeSymbol);
+                case SymbolKind.Namespace:
+                    return SameNamespace(s1 as INamespaceSymbol, s2 as INamespaceSymbol);
+                case SymbolKind.Assembly:
+                    return SameAssembly(s1 as IAssemblySymbol, s2 as IAssemblySymbol);
+                case SymbolKind.NetModule:
+                    return SameModule(s1 as IModuleSymbol, s2 as IModuleSymbol);
+                default:
+                    return false;
+            }
+        }
+
+        private static bool SameModule(IModuleSymbol m1, IModuleSymbol m2)
+        {
+            Debug.Assert(m1 != null);
+            return m2 != null && SameOriginalSymbol(m1.ContainingSymbol, m2.ContainingSymbol);
```

Comment:
modules have names (i think...).  Do those need to match?  If not, can you put a comment that it is intentional that we are not checking them?

Reply:
I don't think it can make any difference, since modules are merged into the assembly and cannot have conflicting members.


---
In reply to: [184568222](https://github.com/dotnet/roslyn/pull/26367#discussion_r184568222) [](ancestors = 184568222)